deepfence_worker/ingesters/common.go

```diff
@@ -36,69 +36,31 @@ func CommitFuncStatus[Status any](ts utils.Neo4jScanType) func(ns string, data [
 		}
 		defer tx.Close()
 
-		var in_progress_query, others_query string
-		switch ts {
-		default:
-			in_progress_query = `
-			UNWIND $batch as row
-			MATCH (n:` + string(ts) + `{node_id: row.scan_id})
-			WHERE NOT n.status IN $cancel_states
-			SET n.status = row.scan_status, n.status_message = row.scan_message, n.updated_at = TIMESTAMP()
-			WITH n
-			OPTIONAL MATCH (n) -[:DETECTED]- (m)
-			WITH n, count(m) as count
-			MATCH (n) -[:SCANNED]- (r)
-			SET r.` + ingestersUtil.ScanCountField[ts] + `=count, r.` + ingestersUtil.ScanStatusField[ts] + `=n.status, r.` + ingestersUtil.LatestScanIdField[ts] + `=n.node_id`
-
-			others_query = `
-			UNWIND $batch as row
-			MATCH (n:` + string(ts) + `{node_id: row.scan_id})
-			SET n.status = row.scan_status, n.status_message = row.scan_message, n.updated_at = TIMESTAMP()
-			WITH n
-			OPTIONAL MATCH (n) -[:DETECTED]- (m)
-			WITH n, count(m) as count
-			MATCH (n) -[:SCANNED]- (r)
-			SET r.` + ingestersUtil.ScanCountField[ts] + `=count, r.` + ingestersUtil.ScanStatusField[ts] + `=n.status, r.` + ingestersUtil.LatestScanIdField[ts] + `=n.node_id`
-
-		case utils.NEO4J_CLOUD_COMPLIANCE_SCAN:
-			in_progress_query = `
-			UNWIND $batch as row
-			MATCH (n:` + string(ts) + `{node_id: row.scan_id})
-			WHERE NOT n.status IN $cancel_states
-			SET n.status = row.scan_status, n.status_message = row.scan_message, n.updated_at = TIMESTAMP()
-			WITH n
-			OPTIONAL MATCH (n) -[:DETECTED]- (m)
-			WITH n, count(m) as total_count
-			OPTIONAL MATCH (n) -[:DETECTED]- (m)
-			WITH  n, total_count, m.resource as arn, count(m) as count
-			OPTIONAL MATCH (n) -[:SCANNED]- (cn) -[:OWNS]- (cr:CloudResource{arn: arn})
-			SET cn.` + ingestersUtil.ScanCountField[ts] + `=total_count, cn.` + ingestersUtil.ScanStatusField[ts] + `=n.status, cn.` + ingestersUtil.LatestScanIdField[ts] + `=n.node_id
-			SET cr.` + ingestersUtil.ScanCountField[ts] + `=count, cr.` + ingestersUtil.ScanStatusField[ts] + `=n.status, cr.` + ingestersUtil.LatestScanIdField[ts] + `=n.node_id`
-
-			others_query = `
-			UNWIND $batch as row
-			MATCH (n:` + string(ts) + `{node_id: row.scan_id})
-			SET n.status = row.scan_status, n.status_message = row.scan_message, n.updated_at = TIMESTAMP()
-			WITH n
-			OPTIONAL MATCH (n) -[:DETECTED]- (m)
-			WITH n, count(m) as total_count
-			OPTIONAL MATCH (n) -[:DETECTED]- (m)
-			WITH  n, total_count, m.resource as arn, count(m) as count
-			OPTIONAL MATCH (n) -[:SCANNED]- (cn) -[:OWNS]- (cr:CloudResource{arn: arn})
-			SET cn.` + ingestersUtil.ScanCountField[ts] + `=total_count, cn.` + ingestersUtil.ScanStatusField[ts] + `=n.status, cn.` + ingestersUtil.LatestScanIdField[ts] + `=n.node_id
-			SET cr.` + ingestersUtil.ScanCountField[ts] + `=count, cr.` + ingestersUtil.ScanStatusField[ts] + `=n.status, cr.` + ingestersUtil.LatestScanIdField[ts] + `=n.node_id`
-		}
+		query := `
+		UNWIND $batch as row
+		OPTIONAL MATCH (m) -[:DETECTED]- (n:` + string(ts) + `{node_id: row.scan_id}) -[:SCANNED]- (r)
```

Comment:
We have the node "r" here. Here the "r" can be `ContainerImage` , `Container` etc. 
Now we need to take care of the pod logic if this is `Container` and has `pod_id` field as not null. 
I am not sure if we can implement the logic using the "CASE" statement here and return the pod_id if not null.
Let me know your thoughts on this.

Reply:
This case is still being handled, it is in a separate query (`updatePodScanStatus`)
Later we will do `updateCloudResourceScanStatus` and `updatePodScanStatus` outside the hot path entirely, but for now they are there.